### PR TITLE
Koa

### PR DESCRIPTION
**Pull request by marudor**

### CODE REVIEW

**Comment by marudor** [comment]
The failure of koa-router probably needs some thought.
koa-router depends on koa. But our test runner tests each libdef isolated. (which makes sense for most cases)
We need some way to reference the koa libdef.

File: definitions/koa_v2.x.x/flow_>=v0.23.x/koa_v2.x.x.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+type Koa$Request$query = Object;
+type Koa$Request$querystring = string;
+type Koa$Request$search = string;
+type Koa$Request$socket = any;
+type Koa$Request$stale = bool;
+type Koa$Request$subdomains = string[];
+type Koa$Request$type = string;
+type Koa$Request$url = string;
+
+declare class Koa$Request extends http$ClientRequest {
```
Comment (netproteus) [inline definitions/koa_v2.x.x/flow_>=v0.23.x/koa_v2.x.x.js]: This is wrong Koa$Request doesn't extend anything

File: definitions/koa_v2.x.x/flow_>=v0.23.x/koa_v2.x.x.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+  is(...types: string[]): bool;
+  length: Koa$Request$length;
+  method: Koa$Request$method;
+  origin: Koa$Request$origin;
+  originalUrl: Koa$Request$originalUrl;
+  path: Koa$Request$path;
+  protocol: Koa$Request$protocol;
+  query: Koa$Request$query;
+  querystring: Koa$Request$querystring;
+  search: Koa$Request$search;
```
Comment (netproteus) [inline definitions/koa_v2.x.x/flow_>=v0.23.x/koa_v2.x.x.js]: `req: http$IncomingMessage;
  res: http$ClientRequest;`
Should be included here

File: definitions/koa_v2.x.x/flow_>=v0.23.x/koa_v2.x.x.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+  header: Object;
+  headers: Object;
+  headerSent: bool;
+  is(...types: string[]): bool;
+  lastModified: ?Date;
+  length: ?number;
+  message?: ?string;
+  redirect(url: string, alt?: string): void;
+  remove(field: string): void;
+  set(field: string, value: string): void;
```
Comment (netproteus) [inline definitions/koa_v2.x.x/flow_>=v0.23.x/koa_v2.x.x.js]: `req: http$IncomingMessage;`
`res: http$ClientRequest;`
Should be included here

File: definitions/koa_v2.x.x/flow_>=v0.23.x/koa_v2.x.x.js [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+  search: Koa$Request$search;
+  secure: bool;
+  socket: Koa$Request$socket;
+  stale: Koa$Request$stale;
+  subdomains: Koa$Request$subdomains;
+  type: Koa$Request$type;
+  url: Koa$Request$url;
+}
+
+declare class Koa$Response extends http$IncomingMessage {
```
Comment (netproteus) [inline definitions/koa_v2.x.x/flow_>=v0.23.x/koa_v2.x.x.js]: This is wrong Koa$Response doesn't extend anything

File: definitions/koa_v2.x.x/flow_>=v0.23.x/koa_v2.x.x.js [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    secure?: bool,
+    httpOnly?: bool,
+  }): this;
+}
+declare class Koa$Context {
+  app: Koa$Koa;
+  assert(value: any, msgOrStatus?: number|string, properties?: Object): void;
+  assert(value: any, msgOrStatus?: number|string, statusOrMsg?: number|string, properties?: Object): void;
+  cookies: Koa$Cookies;
+  req: http$ClientRequest;
```
Comment (netproteus) [inline definitions/koa_v2.x.x/flow_>=v0.23.x/koa_v2.x.x.js]: this should be 
`req: http$IncomingMessage;`

File: definitions/koa_v2.x.x/flow_>=v0.23.x/koa_v2.x.x.js [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+  request: Koa$Request;
+  res: http$IncomingMessage;
```
Comment (netproteus) [inline definitions/koa_v2.x.x/flow_>=v0.23.x/koa_v2.x.x.js]: this should be 
`req: http$IClientRequest;`

**Comment by ricardogama** [comment]
@marudor Any updates on this? What about making a separate PR just with koa types and handle the module dependencies issue of the other modules after?

**Comment by marudor** [comment]
@ricardogama Sure, can do that later on.

**Comment by spmsupun** [comment]
Why is this closed ?

**Pull request closed without merging**